1.21.0

matteobruni opened the pull request:
feat: speed decay
feat: confetti shape, closes #605
feat: emitters spawn color
feat: animation loop count for opacity and size

matteobruni committed:
feat: speed decay

ghost commented:
## DeepCode's analysis on [#75f1f6](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/d33a24210f9c1c5ee2f9056ab56105143aa6bdc4/matteobruni/tsparticles/75f1f69b034528f4518c58e58ab1f0453ddec283/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=2&) found:

- :information_source: **2** minor issues. :point_down:
- :heavy_check_mark: **1** issue was fixed.
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">'public' is implicit. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/75f1f69b034528f4518c58e58ab1f0453ddec283/core/main/src/Core/Particle/Updater.ts#L119">Updater.ts:119</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/75f1f69b034528f4518c58e58ab1f0453ddec283/core/main/src/Core/Particle/Updater.ts#L361">Updater.ts:361</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/75f1f69b034528f4518c58e58ab1f0453ddec283/core/main/src/Core/Particle.ts#L42">Particle.ts:42</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/d33a24210f9c1c5ee2f9056ab56105143aa6bdc4/matteobruni/tsparticles/75f1f69b034528f4518c58e58ab1f0453ddec283/pr/_/%2Fcore%2Fmain%2Fsrc%2FCore%2FParticle%2FUpdater.ts/javascript%2Ftslint%2Fmember-access/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">non-arrow functions are forbidden Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/75f1f69b034528f4518c58e58ab1f0453ddec283/shapes/confetti/src/shape.ts#L25">shape.ts:25</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/d33a24210f9c1c5ee2f9056ab56105143aa6bdc4/matteobruni/tsparticles/75f1f69b034528f4518c58e58ab1f0453ddec283/pr/_/%2Fshapes%2Fconfetti%2Fsrc%2Fshape.ts/javascript%2Ftslint%2Fonly-arrow-functions/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/d33a24210f9c1c5ee2f9056ab56105143aa6bdc4/matteobruni/tsparticles/75f1f69b034528f4518c58e58ab1f0453ddec283/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=2&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=matteobruni)

matteobruni committed:
feat: confetti shape, closes #605

matteobruni committed:
Merge branch 'main' into v1

matteobruni committed:
feat: emitters spawn color

matteobruni committed:
Merge branch 'v1' of https://github.com/matteobruni/tsparticles into v1

matteobruni committed:
feat: animation loop count for opacity and size

matteobruni committed:
chore(release): published new version

 - ng-particles@2.4.0
 - inferno-particles@1.4.0
 - jquery-particles@1.21.0
 - preact-particles@1.21.0
 - react-tsparticles@1.21.0
 - svelte-particles@1.4.0
 - particles.vue@2.4.0
 - particles.vue3@1.4.0
 - tsparticles-editor@1.21.0
 - tsparticles@1.21.0
 - ng-particles-demo@1.6.0
 - editor-demo@1.6.0
 - inferno-particles-demo@1.4.0
 - jquery-particles-demo@1.6.0
 - tsparticles-demo@1.6.0
 - preact-particles-demo@1.6.0
 - react-tsparticles-demo@1.6.0
 - svelte-demo@1.4.0
 - particles.vue-demo@1.6.0
 - particles.vue3-demo@1.4.0
 - tsparticles-preset-60fps@1.6.0
 - tsparticles-preset-background-mask@1.6.0
 - tsparticles-preset-basic@1.6.0
 - tsparticles-preset-big-circles@1.6.0
 - tsparticles-preset-bouncing@1.6.0
 - tsparticles-preset-bubbles@1.6.0
 - tsparticles-preset-fire@1.6.0
 - tsparticles-preset-font-awesome@1.6.0
 - tsparticles-preset-fountain@1.6.0
 - tsparticles-preset-snow@1.6.0
 - tsparticles-preset-stars@1.6.0
 - tsparticles-shape-bubble@1.6.0
 - tsparticles-shape-confetti@1.6.0
 - tsparticles-shape-heart@1.6.0
 - tsparticles-shape-multiline-text@1.6.0
 - tsparticles-shape-rounded-rect@1.6.0
 - tsparticles-shape-spiral@1.6.0
 - cra-template-particles-typescript@1.4.0
 - cra-template-particles@1.4.0
 - tsparticles-website@1.21.0
